editors/code/src/main.ts

```diff
@@ -333,3 +352,53 @@ async function getServer(config: Config, state: PersistentState): Promise<string
     await state.updateServerVersion(config.package.version);
     return dest;
 }
+
+async function performDownloadWithRetryDialog<T>(downloadFunc: () => Promise<T>, state: PersistentState): Promise<T> {
```

Comment:
```suggestion
async function downloadWithRetryDialog<T>(state: PersistentState, downloadFunc: () => Promise<T>): Promise<T> {
```
The verbs `perform` and `do` seem redundant, I also recommend passing callbacks as the last parameter to be able to invoke it with a closure more cleanly